llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -1520,10 +1537,10 @@ void HWAddressSanitizer::sanitizeFunction(Function &F,
                    Mapping.WithFrameRecord &&
                    !SInfo.AllocasToInstrument.empty());
 
+  DominatorTree &DT = FAM.getResult<DominatorTreeAnalysis>(F);
```

Comment:
This now may request the DT/LI/PDT in cases where it wasn't before, which should probably avoided. In cases they are only needed for updating them/preserving them, you can use `FAM.getCachedResult`, which won't cause the DT to be computed if it isn't already cached.